lib/bloc/fee_options/fee_options_bloc.dart

```diff
@@ -34,39 +38,57 @@ class FeeOptionsBloc extends Cubit<FeeOptionsState> {
   }
 
   Future<List<FeeOption>> _constructFeeOptionList(
-    String toAddress,
-    RecommendedFees recommendedFees,
-  ) async {
+      {required String toAddress, required RecommendedFees recommendedFees, String? swapAddress}) async {
     final List<FeeOption> feeOptions = [
       FeeOption(
         processingSpeed: ProcessingSpeed.economy,
         waitingTime: const Duration(minutes: 60),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.hourFee),
+        fee: swapAddress == null
+            ? await _calculateSweepTransactionFee(toAddress: toAddress, satPerVbyte: recommendedFees.hourFee)
+            : await _calculateRefundTransactionFee(
+                satPerVbyte: recommendedFees.hourFee, toAddress: toAddress, swapAddress: swapAddress),
         feeVByte: recommendedFees.hourFee,
       ),
       FeeOption(
         processingSpeed: ProcessingSpeed.regular,
         waitingTime: const Duration(minutes: 30),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.halfHourFee),
+        fee: swapAddress == null
+            ? await _calculateSweepTransactionFee(
+                toAddress: toAddress, satPerVbyte: recommendedFees.halfHourFee)
+            : await _calculateRefundTransactionFee(
+                satPerVbyte: recommendedFees.halfHourFee, toAddress: toAddress, swapAddress: swapAddress),
         feeVByte: recommendedFees.halfHourFee,
       ),
       FeeOption(
         processingSpeed: ProcessingSpeed.priority,
         waitingTime: const Duration(minutes: 10),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.fastestFee),
+        fee: swapAddress == null
+            ? await _calculateSweepTransactionFee(
+                toAddress: toAddress, satPerVbyte: recommendedFees.fastestFee)
+            : await _calculateRefundTransactionFee(
+                satPerVbyte: recommendedFees.fastestFee, toAddress: toAddress, swapAddress: swapAddress),
         feeVByte: recommendedFees.fastestFee,
       ),
     ];
     emit(state.copyWith(feeOptions: feeOptions));
     return feeOptions;
   }
 
-  Future<int> _calculateTransactionFee(String toAddress, int satsPerVbyte) async {
-    final req = PrepareSweepRequest(
-      satPerVbyte: satsPerVbyte,
-      toAddress: toAddress,
-    );
-    final response = await _breezSDK.prepareSweep(req: req);
+  Future<int> _calculateSweepTransactionFee({required String toAddress, required int satPerVbyte}) async {
+    final response = await _breezSDK.prepareSweep(
+        req: PrepareSweepRequest(toAddress: toAddress, satPerVbyte: satPerVbyte));
     return response.sweepTxFeeSat;
   }
+
+  Future<int> _calculateRefundTransactionFee(
+      {required toAddress, required int satPerVbyte, required String swapAddress}) async {
+    final response = await _breezSDK.prepareRefund(
+      req: PrepareRefundRequest(
+        swapAddress: swapAddress,
+        toAddress: toAddress,
+        satPerVbyte: satPerVbyte,
+      ),
```

Comment:
Extract to a variable for debugging purposes and cleaner code.

Reply:
done